src/node.cc

```diff
@@ -2446,6 +2455,31 @@ void ProcessEmitWarning(Environment* env, const char* fmt, ...) {
   f.As<v8::Function>()->Call(process, 1, &arg);
 }
 
+void ProcessEmitDeprecationWarning(Environment* env,
+                                   const char* warning,
+                                   const char* deprecation_code) {
+  HandleScope handle_scope(env->isolate());
+  Context::Scope context_scope(env->context());
+
+  Local<Object> process = env->process_object();
+  MaybeLocal<Value> emit_warning = process->Get(env->context(),
+      FIXED_ONE_BYTE_STRING(env->isolate(), "emitWarning"));
+
+  Local<Value> args[3];
+  args[0] = node::OneByteString(env->isolate(), warning);
+  args[1] = FIXED_ONE_BYTE_STRING(env->isolate(), "DeprecationWarning");
+  args[2] = node::OneByteString(env->isolate(), deprecation_code);
+
+  Local<Value> f;
+
+  if (!emit_warning.ToLocal(&f)) return;
```

Comment:
If `emit_warning` is empty, we should probably return immediately after that. Creating handles to strings in between the failure and returning is probably no big deal, but at some point somebody might think it’s okay to slip in code that potentially calls back into JS instead?

Either way, one issue here is that `ProcessEmitDeprecationWarning` has no way of signaling its caller whether it leaves an exception pending or not (i.e. if `emit_warning` is empty or the `->Call()` below returned an empty result).

I think the options for that are:

- Return `Maybe<bool>`, like the V8 API does, which is `Just(true)` on success and `Nothing<bool>()` on failure, and leave it for the caller to handle the issue.
- Use `.ToLocalChecked()` on everything. The big downside is that emitting deprecation warnings with `--throw-deprecation` on the command line *does* throw an error as part of regular operations.